crates/burn-import/src/burn/node/slice.rs

```diff
@@ -32,37 +39,69 @@ impl<PS: PrecisionSettings> NodeCodegen<PS> for SliceNode {
             None => quote! { .. },
         });
 
-        quote! {
-            let #output = #input.slice(burn::tensor::s![#(#ranges),*]);
+        match &self.input {
+            Type::Tensor(tensor) => {
+                let input = scope.tensor_use_owned(tensor, node_position);
+                quote! {
+                    let #output = #input.slice(s![#(#ranges),*]);
+                }
+            }
+            Type::Shape(shape) => {
+                let shape_len = shape.rank;
+                let shape_len = Literal::usize_unsuffixed(shape_len); // Use unsuffixed literal
+                let shape = shape.name.clone();
+
+                // first item from the range
+                let mut ranges = ranges;
+                let range = ranges.next().unwrap();
+
+                quote! {
+                    let #output: [usize; #output_rank] = #shape[s![#range].into_ranges([#shape_len].into())[0].clone()].try_into().unwrap();
+
+                }
+            }
+            _ => {
+                panic!("Unsupported input type for SliceNode");
+            }
         }
     }
     fn into_node(self) -> Node<PS> {
         Node::Slice(self)
     }
-}
 
+    fn register_imports(&self, imports: &mut BurnImports) {
+        imports.register("burn::tensor::s");
+        match &self.input {
+            Type::Shape(_) => {
+                imports.register("burn::tensor::RangesArg");
+            }
+            _ => {}
+        }
+    }
+}
 #[cfg(test)]
 mod tests {
     use burn::record::FullPrecisionSettings;
 
     use super::*;
     use crate::burn::{
-        TensorType,
+        ShapeType, TensorType,
         graph::BurnGraph,
         node::{slice::SliceNode, test::assert_tokens},
     };
 
     #[test]
-    fn test_codegen_slice() {
+    fn test_codegen_slice_tensor() {
         let mut graph = BurnGraph::<FullPrecisionSettings>::default();
         graph.register(SliceNode::new(
-            TensorType::new_float("tensor1", 4),
-            TensorType::new_float("tensor2", 4),
-            vec![Some((0, 1)), Some((0, 1)), Some((0, 1)), Some((0, 1))],
+            Type::Tensor(TensorType::new_float("tensor1", 4)),
+            Type::Tensor(TensorType::new_float("tensor2", 4)), // Output rank changes
```

Comment:
```suggestion
            Type::Tensor(TensorType::new_float("tensor2", 4)),
```

Guess the comment is no longer valid